README.md

```diff
@@ -95,6 +95,55 @@ However, here are some rules that we'd like to emphasize since they are not visi
+|Expected Difficulty      | Medium |
+|Expected Time Commitment | Large project - 350 hours |
+|Mentor                   | Wojciech Mazur (GitHub: [@WojciechMazur](https://github.com/Wojciech Mazur), Email: [wmazur@virtuslab.com](wmazur@virtuslab.com))|
+|Co-mentor                | TODO |
+
+| Title                   | JMH-Compliant Benchmarking Framework for Scala Native & Scala.js |
+|-------------------------|-------------------------------------------------------------|
+|Link to Project          | [Scala Native](https://scala-native.org/) / [Scala.js](https://www.scala-js.org/) |
+|Brief Description        | This project aims to implement a benchmarking framework similar to JMH (Java Microbenchmark Harness)  allowing for accurate and reliable performance measurements on non JVM platforms. Both Scala Native and Scala.js cannot consume a modified JVM bytecode emitted by JMH framework. The goal is to create a runtime implementation for executing microbenchmarks and a Scala compiler plugin performing required transformations of Scala.js / Scala Native code based on JMH framework compiletime annotations. |
+|Expected Outcome         | A benchmarking framework that mimics JMH APIs and produces reliable results for Scala Native and Scala.js. Demonstration benchmarks showcasing usage. |
+|Prerequisites            | Good understanding of benchmarking principles, Scala Native, Scala.js, and Scala compiler plugins. Some knowledge of JMH is beneficial. |
+|Expected Difficulty      | Medium |
+|Expected Time Commitment | Medium project - 175 hours |
+|Mentor                   | Wojciech Mazur (GitHub: [@WojciechMazur](https://github.com/Wojciech Mazur), Email: [wmazur@virtuslab.com](wmazur@virtuslab.com))|
+|Co-mentor                | TODO |
+
+### Scala Native Projects
+
+| Title                   | WebAssembly Backend for Scala Native |
+|-------------------------|--------------------------------------|
+|Link to Project          | [Scala Native](https://scala-native.org/) |
+|Brief Description        | This project aims to introduce WebAssembly (Wasm) as a new backend for Scala Native, allowing Scala Native programs to run efficiently in WebAssembly-supported environments, such serverless runtimes (via WASI) or browsers. The goal is to design an alternative to current LLVM-based backend allowing to emit Wasm directlly |
+|Expected Outcome         | A functional WebAssembly backend capable of compiling Scala Native programs to Wasm, with basic runtime support and test cases demonstrating correctness. |
+|Prerequisites            | Strong knowledge of Scala and compilers. Familiarity with LLVM, WebAssembly, and Scala Native’s internals is a plus. |
+|Expected Difficulty      | Hard |
+|Expected Time Commitment | Large project - 350 hours |
+|Mentor                   |Wojciech Mazur (GitHub: [@WojciechMazur](https://github.com/Wojciech Mazur), Email: [wmazur@virtuslab.com](wmazur@virtuslab.com))|
+|Co-mentor                | TODO |
```

Comment:
@tanishiking would you be interested in co-mentoring? Might be an interesting experience to look at the GSOC from the perspective of mentor instead of mentoree

Reply:
Wouldn't that be duplicate work, compared to @tanishiking's current efforts to support the Wasm Component Model within the Scala.js pipeline (https://github.com/scala-js/scala-js/issues/5121 / https://github.com/scala-wasm/scala-wasm/pull/5)?

Unless you make a point of focusing on the multithreading support of Scala Native. WasmGC doesn't support threading yet, but it can be done with the linear memory.

It seems like a *huge* project to me, regardless. Consider that @tanishiking and I certainly put more than 350 hours *each* on making Scala.js-on-WasmGC work. A Native backend for linear-memory Wasm would likely be an even bigger investment.

Reply:
Hmm, makes sense. In theory we can allow for SN-WASM be executed only in single-threaded mode (we're having a link-time switch for that) , but probably that might be an overkill now. 
I know that we can build WebAssembly using LLVM (I have a PoC branch) but the problem is GC and Exception Handling. Based on that an alternative project might be adaptation of current Immix GC to use shadow-stack when bulding for WASM

Reply:
Thanks for the invite!
But I’ve also been talking with @armanbilge about co-mentoring on PoC WASIp2 support for libraries like Cats Effect and FS2, so I'd like to focus on Wasm/WASI via Scala.js-WasmGC for now. 🙏

Also, as you mentioned, GC and Exception would be a big challenge if we go with LLVM. Even if we implement it using a shadow stack, there'd be a overhead and potential cycle collection issues. I'm not sure that's the right direction for Wasm (even if we can re-use most of LLVM/C ecosystem).

(off topic: Utilizing https://popl25.sigplan.org/details/waw-2025-papers/8/An-MLIR-Dialect-for-WebAssembly would be interesting, but this would be a one research topic rather than GSoC project, and we might end up not re-using LLVM and C ecosystem)